src/crypto/x448.rs

```diff
@@ -0,0 +1,261 @@
+use hkdf::Hkdf;
+use log::debug;
+use rand::{CryptoRng, Rng};
+use sha2::Sha512;
+use zeroize::{Zeroize, ZeroizeOnDrop, Zeroizing};
+
+use crate::crypto::{aes_kw, Decryptor, KeyParams};
+use crate::errors::Result;
+use crate::types::{PlainSecretParams, PublicParams};
+
+/// Secret key for X448
+#[derive(Clone, derive_more::Debug, PartialEq, Eq, Zeroize, ZeroizeOnDrop)]
+pub struct SecretKey {
+    #[debug("..")]
+    pub(crate) secret: [u8; 56],
+}
+
+impl KeyParams for SecretKey {
+    type KeyParams = ();
+
+    fn key_params(&self) {}
+}
+
+impl Decryptor for SecretKey {
+    // - Ephemeral X448 public key (56 bytes).
+    // - Recipient public key (56 bytes). [FIXME: use PublicParams and expect PublicParams::X448?]
+    // - Encrypted and wrapped session key.
+    type Data<'a> = ([u8; 56], [u8; 56], &'a [u8]);
```

Comment:
would be nicer to make this at least a struct

Reply:
I refactored this a bit, but am very unsure what names I'd like, and if this is the structure we want: b877594112607c5c52a3aa9a88269fe035879247

Reply:
structure looks good to me